### PR TITLE
chore(next-release/main): Merge latest from main

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes:

- Pulls in the latest changes from main (including the bump to next.js version and all changes outside of pages)
- Merges in content changes

I think our last merge from main missed pulling in the index files in pages-old that look like the below (the code that includes the ChooseFilterPage component), so there are quite a few changes in that pages-old directory. I left them as is since we're not rendering those pages and they are just kind of there for reference for now.

```
import ChooseFilterPage from '@/components/ChooseFilterPage';

import { INTEGRATION_FILTER_OPTIONS } from '@/utils/filter-data.ts';

<ChooseFilterPage
  directoryPath="/ChooseFilterPage"
  address="/guides/api-graphql/graphql-pagination"
  filterKind="platform"
  filters={['js', 'android', 'ios']}
  currentFilter="all"
  message={'Choose a platform:'}
/>
```

I noticed we missed some find/replace of `<InlineFilter filters={["js"]} />` to use `javascript` instead of `js`. Will address in a separate PR.

https://mergemain1106.d2bfwhpcsj9awv.amplifyapp.com/

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -148,13 +147,14 @@
...
+    "build": "node tasks/generate-sitemap.mjs && next build",
     "build:release": "yarn build && next-image-export-optimizer --exportFolderPath client/www/next-build",
     "next-build": "next build",
     "next-start": "next start",
     "prepare": "husky install",
     "analyze": "ANALYZE=true yarn next-build",
     "prebuild": "node src/directory/generateDirectory.mjs && node src/directory/generateFlatDirectory.mjs",
-    "migrate": "node tasks/new_ia_migration.ts"
+    "migrate": "node tasks/new_ia_migration.ts",
+    "amplify:submissionsLambda": "cd amplify/backend/function/submissionsLambda/src && yarn install && yarn build && yarn test"
```
Comment (timngyn) [inline package.json:158]: We can remove this `amplify:submissionsLambda` script since we no longer use it

**Review by timngyn — COMMENTED** [review]
Thanks for doing this @hbuchel! Just had one comment

File: package.json [hunk 2]
```diff
@@ -86,9 +86,8 @@
     "husky": "^8.0.1",
     "jest": "^26.6.3",
     "jest-cli": "^26.6.3",
-    "json5": "^2.2.3",
     "next-bundle-analyzer": "^0.6.7",
-    "prettier": "^3.0.3",
+    "prettier": "^1.19.1",
```
Comment (jacoblogan) [inline package.json:90]: don't have the full context here but seems strange that we would be downgrading this version

Comment (hbuchel) [reply package.json]: Fixed in latest, thanks for the catch!

File: package.json (hunk 1 above)
Comment (hbuchel) [reply package.json]: Good catch, thanks! Removed in latest.

**Review by reesscot — APPROVED** [review]
LGTM

**Review by sowmyadev-kota — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Review by jacoblogan — APPROVED** [review]

**Pull request merged**